ydb/tests/functional/backup_collection/basic_user_scenarios.py

```diff
@@ -1434,3 +1434,177 @@ def q(role: str) -> str:
+        res = self._execute_yql(create_restore_sql)
+        assert res.exit_code == 0, f"CREATE restore collection {coll_restore} failed"
+        self.wait_for_collection(coll_restore, timeout_s=30)
+
+        # import exported snapshots with ts <= target_ts
+        all_dirs = sorted([d for d in os.listdir(export_dir) if os.path.isdir(os.path.join(export_dir, d))])
+        chosen = [d for d in all_dirs if d.split("_", 1)[0] <= target_ts]
+        assert chosen, f"No exported snapshots with ts <= {target_ts} found in {export_dir}: {all_dirs}"
+
+        for name in chosen:
+            src = os.path.join(export_dir, name)
+            dest_path = f"/Root/.backups/collections/{coll_restore}/{name}"
+            r = yatest.common.execute(
+                [
+                    backup_bin(),
+                    "--verbose",
+                    "--endpoint",
+                    "grpc://localhost:%d" % self.cluster.nodes[1].grpc_port,
+                    "--database",
+                    self.root_dir,
+                    "tools",
+                    "restore",
+                    "--path",
+                    dest_path,
+                    "--input",
+                    src,
+                ],
+                check_exit_code=False,
+            )
+            out = (r.std_out or b"").decode("utf-8", "ignore")
+            err = (r.std_err or b"").decode("utf-8", "ignore")
+            assert r.exit_code == 0, f"tools restore import failed for {name}: stdout={out} stderr={err}"
+
+        # wait until imported snapshots registered in scheme
+        deadline = time.time() + 60
+        expected = set(chosen)
+        while time.time() < deadline:
+            kids = set(self.get_collection_children(coll_restore))
+            if expected.issubset(kids):
+                break
+            time.sleep(1)
+        else:
+            raise AssertionError(f"Imported snapshots did not appear in collection {coll_restore} within timeout. Expected: {sorted(chosen)}")
+
+        self._drop_tables([t_orders, t_products])
+
+        res_restore = self._execute_yql(f"RESTORE `{coll_restore}`;")
+        assert res_restore.exit_code == 0, f"RESTORE failed: {getattr(res_restore, 'std_err', None) or getattr(res_restore, 'std_out', None)}"
+
+        expected_orders = snapshot_rows[target_snap]["orders"]
+        expected_products = snapshot_rows[target_snap]["products"]
+
+        self._verify_restored_table_data(t_orders, expected_orders)
+        self._verify_restored_table_data(t_products, expected_products)
+
+        coll_present = self.collection_exists(collection_src)
+
+        if not coll_present:
+            logger.info("Starting collection %s not present (deleted) — OK", collection_src)
+        else:
+            logger.info(
+                f"Starting collection {collection_src} is present and incremental backups appear enabled. "
+                "Expected: starting collection removed OR incremental backups disabled."
+            )
```

Comment:
Huge, more than one screen with inline functions. It's better to break on named phases and define reusable functions outside